server/src/main/java/org/elasticsearch/transport/RemoteClusterConnection.java

```diff
@@ -236,13 +206,25 @@ boolean isNodeConnected(final DiscoveryNode node) {
      * Get the information about remote nodes to be rendered on {@code _remote/info} requests.
      */
     public RemoteConnectionInfo getConnectionInfo() {
-        return new RemoteConnectionInfo(
+        if (connectionStrategy instanceof SniffConnectionStrategy) {
```

Comment:
instead of instanceof, can we call a method on the `connectionStrategy` object here?

Reply:
I would like to work on this in follow-up. The `RemoteConnectionInfo` piece of work kind of needs it's own refactoring. It currently is tightly coupled to the "sniff" strategy and cross cluster search (the skip unavailable boolean). In the current PR, it retains the existing behavior.